cllc-public-app/Controllers/AccountController.cs

```diff
@@ -101,65 +113,101 @@ public async Task<IActionResult> GetCurrentBCeIDBusiness()
-                    entity.shareholderFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Central Securities Register").Result;
-                    var tiedHouse = _dynamicsClient.AdoxioTiedhouseconnections
-                        .Get(filter: $"_adoxio_accountid_value eq {entity.Account.id}")
-                        .Value.FirstOrDefault();
-                    if (tiedHouse != null)
-                    {
-                        entity.TiedHouse = tiedHouse.ToViewModel();
-                    }
-                    entity.ChildEntities = GetLegalEntityChildren(entity.AdoxioLegalEntity.id);
-                    return entity;
-                })
-                .ToList();
+            try
+            {
+                legalEntities = _dynamicsClient.Adoxiolegalentities.Get(filter: entityFilter, expand: expandList).Value
+                        .Select(le =>
+                        {
+                            var legalEntity = le.ToViewModel();
+                            var entity = new ViewModels.LegalEntity
+                            {
+                                AdoxioLegalEntity = legalEntity,
+                                Account = le.AdoxioShareholderAccountID == null ? account : le.AdoxioShareholderAccountID.ToViewModel(),
+                            };
+                            entity.corporateDetailsFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Corporate Information").Result;
+                            entity.organizationStructureFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Organization Structure").Result;
+                            entity.keyPersonnelFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Key Personnel").Result;
+                            entity.financialInformationFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Financial Information").Result;
+                            entity.shareholderFilesExists = FileUploadExists(entity.Account.id, entity.Account.name, "Central Securities Register").Result;
+                            var tiedHouse = _dynamicsClient.AdoxioTiedhouseconnections
+                                .Get(filter: $"_adoxio_accountid_value eq {entity.Account.id}")
+                                .Value.FirstOrDefault();
+                            if (tiedHouse != null)
+                            {
+                                entity.TiedHouse = tiedHouse.ToViewModel();
+                            }
+                            entity.ChildEntities = GetLegalEntityChildren(entity.AdoxioLegalEntity.id);
+                            return entity;
+                        })
+                        .ToList();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(LoggingEvents.Error, "Error getting legal entities for the account {accountId}.");
```

Comment:
Can you please change this to use the ODATA error exception rather than a generic exception.  That way we can log full diagnostic information on the problem.